client/core/trade.go

```diff
@@ -283,6 +283,14 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 		if trade.Sell {
 			feeRateSwap = msgMatch.FeeRateBase
 		}
+		log.Infof("Starting negotiation for match %v for order %v with feeRateSwap = %v",
+			mid, oid, feeRateSwap)
+		if feeRateSwap == 0 {
+			// The server is malfunctioning.
+			feeRateSwap = t.wallets.fromWallet.FeeRateWithFallback()
+			log.Warnf("The server provided a 0 swap fee rate for %q! Using %v instead.",
+				t.wallets.fromAsset.Symbol, feeRateSwap)
+		}
```

Comment:
If the server were not recompiled after the recent fee rate changes in https://github.com/decred/dcrdex/pull/505, both `FeeRateBase` and `FeeRateQuote` would be 0.